### PR TITLE
[BugFix] Fix parquet column name case sensitive problem in min-max filter

### PR DESCRIPTION
**Pull request by Smith-Cruise**
## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

If the user query SQL: `SELECT * FROM tbl WHERE Bb=1;` it will return empty, because of min-max filter does not handle column name's case-sensitive properly.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Review by letian-jiang — DISMISSED** [review]

File: be/src/formats/parquet/file_reader.cpp [hunk 1]
```diff
@@ -516,7 +516,8 @@ const tparquet::ColumnMetaData* FileReader::_get_column_meta(const tparquet::Row
                 return &column.meta_data;
             }
         } else {
-            if (boost::algorithm::to_lower_copy(column.meta_data.path_in_schema[0]) == col_name) {
+            if (boost::algorithm::to_lower_copy(column.meta_data.path_in_schema[0]) ==
```
Comment (letian-jiang) [inline be/src/formats/parquet/file_reader.cpp:519]: `boost::algorithm::iequals`

Comment (dirtysalt) [reply be/src/formats/parquet/file_reader.cpp:519]: agree.

Comment (Smith-Cruise) [reply be/src/formats/parquet/file_reader.cpp]: fixed

**Review by letian-jiang — APPROVED** [review]

**Review by stephen-shelby — APPROVED** [review]

**Comment by Smith-Cruise** [comment]
@Mergify backport branch-2.4

**Comment by Smith-Cruise** [comment]
@mergify backport branch-2.3

**Comment by Smith-Cruise** [comment]
@mergify backport branch-2.2

**Pull request merged**